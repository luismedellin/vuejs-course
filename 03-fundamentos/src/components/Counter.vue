<template>
  <h2>{{title || 'Counter'}}</h2>
  <h2>{{customTitle}} - {{start}}</h2>
  <p>{{ counter }} <sup>2</sup> = {{ squareCounter }}</p>
  <button @click="decrease">-1</button>
  <button @click="update(-1)">-1</button>
  <button @click="increase">+1</button>
  <button @click="update(+1)">+1</button>
</template>

<script>
export default {
  props: {
    title: String,
    start: {
      type: Number,
      default: 1,
      validator(value) {
        return value >= 0
      }
      // required: true
    }
  },
  data() {
    return {
      counter: this.start
    }
  },
  methods: {
    getSqureValue() {
      return this.counter * this.counter
    },
    increase() {
      this.counter++;
    },
    decrease() {
      this.counter--;
    },
    update(value) {
      this.counter += value
    }
  },
  computed: {
    squareCounter() {
      return this.counter*this.counter
    },
    customTitle() {
      return this.title || "Counter"
    }
  }
}
</script>

<style>

button {
  background-color: #64B6B7;
  border-radius: 5px;
  border: 1px solid white;
  color: white;
  cursor: pointer;
  margin: 0 5px;
  padding: 5px 15px;
  transition: 0.3s ease-in-out;
}

button: hover {
  background-color: #5aa67b;
  transition: 0.3s ease-in-out;
}

</style>