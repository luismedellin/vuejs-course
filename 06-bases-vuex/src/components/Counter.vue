<template>
    <h1>Counter vuex</h1>
    <h2>Direct access: {{ $store.state.count }}</h2>
    <h2>Computed: {{ countComputed }}</h2>

    <button @click="increment">+1</button>
    <button @click="incrementBy(5)">+5</button>
    <button>Random</button>

    <h1>mapState</h1>
    <h2>mapState: {{ count }}</h2>
    <h2>mapState: {{ lastMutation }}</h2>
</template>

<script>
import { mapState } from 'vuex'

export default {
    // computed: mapState(['count'])
    computed: {
        countComputed() {
            return this.$store.state.count
        },
        ...mapState(['count', 'lastMutation'])
        // ...mapState({
        //     count: state => state.count,
        //     lastMutation: state => state.lastMutation
        // })
    },
    methods: {
        increment() {
            this.$store.commit('increment')
        },
        incrementBy(value) {
            this.$store.commit('incrementBy', value)
        }
    }
}
</script>