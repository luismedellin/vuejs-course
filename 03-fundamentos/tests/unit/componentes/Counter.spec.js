import { shallowMount, mount } from  '@vue/test-utils'
import Counter from '@/components/Counter'

describe('Counter component', () => {
    test('debe de hacer match con el snapshot', () => { 

        const wrapper = shallowMount(Counter)

        expect (wrapper.html()).toMatchSnapshot()
     })
})