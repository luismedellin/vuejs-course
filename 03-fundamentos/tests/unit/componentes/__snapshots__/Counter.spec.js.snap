// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component debe de hacer match con el snapshot 1`] = `
<h2>Counter</h2>
<h2>Counter - 1</h2>
<p>1 <sup>2</sup> = 1</p>
<div class="buttons-container"><button>-1</button><button>-1</button><button>+1</button><button>+1</button></div>
`;
